server/config/config.go

```diff
@@ -991,6 +993,25 @@ func (c *Config) GetZapLogProperties() *log.ZapProperties {
 	return c.logProps
 }
 
+// GetConfigFile gets the config file.
+func (c *Config) GetConfigFile() string {
+	return c.configFile
+}
+
+// RewriteFile rewrites the config file after updating the config.
+func (c *Config) RewriteFile(new *Config) error {
+	filePath := c.GetConfigFile()
+	if filePath == "" {
+		return nil
+	}
+	var buf bytes.Buffer
+	if err := toml.NewEncoder(&buf).Encode(*new); err != nil {
+		return err
+	}
+	// 0644 represent the permission of file which is "-rw-r--r--" here.
+	return ioutil.WriteFile(filePath, buf.Bytes(), 0644)
```

Comment:
To avoid corrupted content during lost power, we should:
1. write the content to temp file
2. sync the temp file
3. rename the temp file to the config file.